anomaly.json

```diff
@@ -0,0 +1,53 @@
+    "description": "Computes anomalies based on normals for a specific temporal frequency.",
+    "categories": [
+        "climatology",
+        "cube",
+        "math"
+    ],
+    "parameters": [
+        {
+            "name": "data",
+            "description": "A data cube with exactly one temporal dimension and the following dimension labels for the given frequency (`YYYY` = four-digit year, `MM` = two-digit month, `DD` two-digit day of month):\n\n* `hourly`: `YYYY-MM-DD-0` - `YYYY-MM-DD-23`\n* `daily`: `YYYY-1` - `YYYY-365`\n* `weekly`: `YYYY-1` - `YYYY-52`\n* `monthly`: `YYYY-1` - `YYYY-12`\n* `yearly`: `YYYY`\n* `single_period` / `climatology_period`: A single dimension label with any name is expected.\n* `seasons`: `YYYY-djf` (December - February), `YYYY-mam` (March - May), `YYYY-jja` (June - August), `YYYY-son` (September - November)\n* `tropical_seasons`: `YYYY-ndjfma` (November - April), `YYYY-mjjaso` (May - October)\n\n``aggregate_temporal_frequency()`` can compute such a data cube.",
+            "schema": {
+                "type": "object",
+                "subtype": "raster-cube"
+            }
+        },
+        {
+            "name": "normals",
+            "description": "A data cube with normals, e.g. daily, monthly or yearly values computed from a process such as ``climatological_normal()``. Must contain exactly one temporal dimension with the following dimension labels for the given frequency:\n\n* `hourly`: `0` - `23`\n* `daily`: `1` - `365`\n* `weekly`: `1` - `52`\n* `monthly`: `1` - `12`\n* `yearly`: Four-digit year numbers\n* `single_period` / `climatology_period`: A single dimension label with any name is expected.\n* `seasons`: `djf` (December - February), `mam` (March - May), `jja` (June - August), `son` (September - November)\n* `tropical_seasons`: `ndjfma` (November - April), `mjjaso` (May - October)",
+            "schema": {
+                "type": "object",
+                "subtype": "raster-cube"
+            }
+        },
+        {
+            "name": "frequency",
+            "description": "Specifies the time intervals available in the data cubes. The following options are available:\n\n* `hourly`: Hour of the day\n* `daily`: Day of the year\n* `weekly`: Week of the year\n* `monthly`: Month of the year\n* `yearly`: Proleptic years\n* `seasons`: Refers to three month periods of the calendar seasons (December - February, March - May, June - August, September - November).\n* `tropical_seasons`: Refers to the six month periods of the tropical seasons (November - April, Mai - October).",
+            "schema": {
+                "type": "string",
+                "enum": [
+                    "hourly",
+                    "daily",
+                    "weekly",
+                    "monthly",
+                    "yearly",
+                    "single_period",
+                    "climatology_period",
+                    "seasons",
+                    "tropical_seasons"
+                ]
```

Comment:
How about adding `decadal` to the list? This could be somehow attractive for climate/weather analysis.

Reply:
I asked @jwagemann about decades, but we couldn't really come up with a good use case, but sure, we can add it if you have a use case in mind? :-)

Reply:
FYI: last weeks I worked on a use case (not openeo based so far)  involving decad based vegetation monitoring. The goal was basically to produce some sort of temporally smoothed NDVI tile each decad based on a larger time window (several months)

Reply:
Added "decade" as frequency to the processes (except climatological_normal), please check.

Reply:
Sorry, in my use case I referring to 10 **day** periods  (three per month: [days 1-10], [days 11-20] and [days 21-31], so 36 per year), and assumed @claxn  was referring to that as well. Compared to the well known "decade" (10 years) it's a more obscure period (I just learned about it couple of weeks ago)  using confusion-prone terminology "dekad" and "dekadal". I have no idea how popular it is and whether it is worth including in the enum.

Reply:
Oh, I thought decades (10 years) were meant. What was your intention @claxn ?

Reply:
My intention was to use it for 10 years for climate science, because there "decadal" is a well-known term. However, we have also had some "decadal" (10 days, exactly like @soxofaan described) use cases in remote sensing, so it seems this is more likely to be used. However, if we use the term "decadal" it could confuse a user. I have googled a bit and found this: 
[10 days](https://english.stackexchange.com/questions/363469/is-there-any-word-for-10-days-in-english) 
How about "tri-monthly"/"trice-monthly"? I think this is very clear then. Should we also include "decadal" (10 years) or drop it? I don't have a use case for it right now...

Reply:
For tri-monthly I'd need to revert the changes to period/frequency and month/monthly.

Are both 10 years and 10 days common use cases we should cover?

I'm thinking whether it would make sense to integrate another parameter that allows to specify an amount (defaulting to 1), e.g.
period = day, amount = 10
period = year, amount = 10
But maybe that makes implementation too complex? Maybe we should just use "10-days" and "decade" or "10-years" as option?

Reply:
FYI: note that with the "10 day" dekads, the last dekad of the month does not always have exactly 10 days: it can range from 8 to 11.

So if you call it a "10-day" period, the user might expect exact 10 day periods, instead of the "monthly aligned, roughly 10 day" periods. But again, this concept is new to me as well, I have no idea how well know/settled all this is.

Reply:
If @jwagemann does not think decadal is urgently needed for climate and meteorological use cases, then I would go for 10 days at the moment. We can add it later if need be. Our use case @ TUW was just to have a 10 day composite of satellite imagery to close some data gaps. 

I prefer tri-month(ly) over 10-day, because for a longer time series a 10 day step is breaking desired temporal boundaries, e.g. end of a month, end of a year etc.

Reply:
True. Best would be if there's a well-known name. tri-monthly also doesn't really tell how things are split. So it depends on good documentation anyway.

Reply:
True, "10-day" is misleading. Best would be if there's a well-known name. tri-monthly also doesn't really tell how things are split. Could even mean "every three months". So it needs good documentation anyway. What I just read is "[decadaily](https://english.stackexchange.com/questions/64086/how-do-you-say-three-times-a-month-in-one-word)". The thread also discusses tri-monthly as confusing word due to every three months vs. month split in 3 parts.

Reply:
I opened issue #150 to track the decadal issue, we may add it in a separate PR. Please keep the discussion going there. Merging without decades for now to get things started.